### ISSUE TITLE
Crash when returning generator from function with option argument

### ISSUE THREAD

**Issue by dapetcu21** [codemix/flow-runtime#72]
This is a:

 - [x] Bug Report
 - [ ] Feature Request
 - [ ] Question
 - [ ] Other

Which concerns:

 - [ ] flow-runtime
 - [x] babel-plugin-flow-runtime
 - [ ] flow-runtime-validators
 - [ ] flow-runtime-mobx
 - [ ] flow-config-parser
 - [ ] The documentation website

---

### What is the current behaviour?

This snippet:

```js
type Options = {};

export const makeApiCallSaga = (opts: Options) => {
  return function* apiCallWatcherSaga (): Generator<*, void, *> {
  };
};
```

Causes the following crash:

```
ERROR in ./src/sagas/api.js
Module build failed: TypeError: /Users/da_petcu21/Work/portal-frontend/src/sagas/api.js: Property id of VariableDeclarator expected node to be of a type ["LVal"] but instead got null
    at Object.validate (/Users/da_petcu21/Work/portal-frontend/node_modules/babel-types/lib/definitions/index.js:109:13)
    at validate (/Users/da_petcu21/Work/portal-frontend/node_modules/babel-types/lib/index.js:505:9)
    at Object.builder (/Users/da_petcu21/Work/portal-frontend/node_modules/babel-types/lib/index.js:466:7)
    at Function (/Users/da_petcu21/Work/portal-frontend/node_modules/babel-plugin-flow-runtime/lib/transformVisitors.js:427:64)
    at NodePath._call (/Users/da_petcu21/Work/portal-frontend/node_modules/babel-traverse/lib/path/context.js:76:18)
    at NodePath.call (/Users/da_petcu21/Work/portal-frontend/node_modules/babel-traverse/lib/path/context.js:48:17)
    at NodePath.visit (/Users/da_petcu21/Work/portal-frontend/node_modules/babel-traverse/lib/path/context.js:105:12)
    at TraversalContext.visitQueue (/Users/da_petcu21/Work/portal-frontend/node_modules/babel-traverse/lib/context.js:150:16)
    at TraversalContext.visitSingle (/Users/da_petcu21/Work/portal-frontend/node_modules/babel-traverse/lib/context.js:108:19)
    at TraversalContext.visit (/Users/da_petcu21/Work/portal-frontend/node_modules/babel-traverse/lib/context.js:192:19)
    at Function.traverse.node (/Users/da_petcu21/Work/portal-frontend/node_modules/babel-traverse/lib/index.js:114:17)
    at NodePath.visit (/Users/da_petcu21/Work/portal-frontend/node_modules/babel-traverse/lib/path/context.js:115:19)
    at TraversalContext.visitQueue (/Users/da_petcu21/Work/portal-frontend/node_modules/babel-traverse/lib/context.js:150:16)
    at TraversalContext.visitMultiple (/Users/da_petcu21/Work/portal-frontend/node_modules/babel-traverse/lib/context.js:103:17)
    at TraversalContext.visit (/Users/da_petcu21/Work/portal-frontend/node_modules/babel-traverse/lib/context.js:190:19)
    at Function.traverse.node (/Users/da_petcu21/Work/portal-frontend/node_modules/babel-traverse/lib/index.js:114:17)
 @ ./src/sagas/login.js 12:11-27
 @ ./src/sagas/index.js
 @ ./src/store/configureStore.dev.js
 @ ./src/store/configureStore.js
 @ ./src/index.js
 @ multi ./tools/webpack-public-path webpack-hot-middleware/client?reload=true babel-polyfill ./src
```

---

### What is the expected behaviour?

It should not crash

---

### Which package versions are you using?

```
├─ babel-plugin-flow-runtime@0.9.1
│  └─ flow-config-parser@^0.3.0
```

**Comment by phpnode** [comment]
released in v0.10.0

**Comment by dapetcu21** [comment]
Cool! Thanks! There were actually some more bugs I think I found, just didn't get to find the MRE and then report them. I'll report them tomorrow.

Marius Petcu,
Web, Mobile (https://petcu.me) and Game Developer (https://critique-gaming.com)
